src/index.js

```diff
@@ -243,26 +244,51 @@ class ChatGPTAuthenticator {
     });
 
     if (response.status === 302) {
-      var splitCookieHeaders = setCookie.splitCookiesString(
-        response.headers.get('set-cookie')
-      );
-      var cookies = setCookie.parse(splitCookieHeaders);
-      this.accessToken = this._getCookie(
+      this.sessionToken = this._getCookie(
         response,
         '__Secure-next-auth.session-token'
       );
       if (
-        typeof this.accessToken === 'undefined' ||
-        this.accessToken === null
+        typeof this.sessionToken === 'undefined' ||
+        this.sessionToken === null
       ) {
-        throw new Error('could not get an accessToken');
+        throw new Error('could not get a sessionToken');
       }
-      return this.accessToken;
+      return this._getAccessToken();
     } else {
       throw new Error('status != 302');
     }
   }
 
+  async _getAccessToken() {
+    const response = await this._fetch(
+      'https://explorer.api.openai.com/api/auth/session',
+      {
+        method: 'GET',
+        credentials: 'include',
+        redirect: 'manual',
+        keepAlive: true,
+        headers: {
+          Host: 'explorer.api.openai.com',
+          Accept: 'application/json',
+          Connection: 'keep-alive',
+          'User-Agent':
+            'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36',
+          'Accept-Language': 'en-GB,en-US;q=0.9,en;q=0.8',
+          cookie: `__Secure-next-auth.session-token=${this.sessionToken}`,
+        },
+      }
+    );
+
+    if (response.status === 200) {
+      const data = await response.json();
+      this.accessToken = data.accessToken;
+      return this.accessToken;
+    } else {
+      throw new Error('status != 200');
+    }
+  }
+
```

Comment:
I did not noticed that I forgot this part 😱 thanks for seeing it. I will add this change it If u don't mind 👍

Reply:
I fixed this part. Thank you @imCorfitz